common/persistence/sql/sqlplugin/visibility.go

```diff
@@ -223,6 +225,25 @@ func getDbFields() []string {
 			dbFields[i] = strcase.ToSnake(f.Name)
 		}
 	}
+
+	// We have to do this for MySQL because the version column has to be the last column in the insert statement.
+	// Otherwise we may do partial updates as the version changes halfway through.
+	// Basically we are just pushing "version" to the end of the list.
+	// We could put "version" at the end of the definition but that would be brittle and break if we add a new column.
+	// This is a hack but we are trapped because MySQL doesn't support row versioning
+	// in a way that prevents out-of-order updates.
+	var foundVersion bool
+	for i := 0; i < len(dbFields)-1; i++ {
+		if dbFields[i] == VersionColumnName {
+			foundVersion = true
+		}
+		if foundVersion {
+			dbFields[i] = dbFields[i+1]
+		}
+	}
+	end := len(dbFields) - 1
+	dbFields[end] = VersionColumnName
```

Comment:
Oh, should we do this for MySQL only in the `buildOnDuplicateKeyUpdate` function?

Reply:
So, I'm not deeply familiar with the `Values(...)` and that might be throwing me off here. Do the orders of the columns need to match in every part of the query? Some part of my very old SQL Experience says yes.

Reply:
We explicitly list the columns when inserting: `INSERT INTO executions_visibility (<columns>) VALUES (<values>)`. The order in `<values>` is the same for `<columns>`.